gemrb/plugins/GUIScript/PythonConversions.cpp

```diff
@@ -137,15 +137,23 @@ PyObject* PyString_FromStringObj(const std::string& s)
 	
 PyObject* PyString_FromStringObj(const String& s)
 {
-	// FIXME: this is wrong, python needs to know the encoding
-	std::string mbstr = MBStringFromString(s);
-	return PyString_FromStringObj(mbstr);
+	return PyUnicode_Decode(reinterpret_cast<const char*>(s.c_str()), s.length() * sizeof(char16_t), "UTF-16", "strict");
 }
 
 String PyString_AsStringObj(PyObject* obj)
 {
-	auto wrap = PyStringWrapper(obj, core->TLKEncoding.encoding.c_str());
-	return StringFromCString(wrap.CString());
+	auto utf16Object = PyUnicode_AsUTF16String(obj);
```

Comment:
seems like it would be cleaner to just take the native (utf-8) data and pass it thought our `StringFromUtf8` function.

Reply:
By native, you mean what Python uses internally? Isn't that somewhat platform/build/version-dependent?